### ISSUE TITLE
Non accessible file reads are not cached

### ISSUE THREAD

**Issue by freshollie** [gajus/get-stack-trace#4]
**TL;DR**: Is there a reason for not caching `fs.accessSync` results during source map resolution? And original line number results from `SourceMapConsumer`?

Considering `get-stack-trace` is used by `slonik` every time an SQL query is made this library attempts to read the source files associated with the call and if they don't exist it will still attempt to read them next call anyway 🤯 .

Recently my team has tracked down a large amount of memory use being attributed to https://github.com/DataDog/dd-trace-js/issues/913#issue-599867608

Digging deeper, it appears the cause is due to `dd-tracer` keeping track of all file reads, and because `get-stack-trace` is constantly trying to read files which don't exist DD is tracking these metrics and filling memory whilst it waits to send the metrics off. When we disabled the DD fs plugin our memory usage dropped significantly during intensive database tasks.

I can see a few improvements which could be made to solve these issues and improve performance of this library:

1. Cache the results of `isReadable`: https://github.com/gajus/get-stack-trace/blob/master/src/resolveCallSiteSourceCodeLocation.js#L36
2. Cache the original line number results of `SourceMapConsumer`: https://github.com/gajus/get-stack-trace/blob/master/src/resolveCallSiteSourceCodeLocation.js#L39

The results of both of these processes should remain static throughout execution, as we don't expect source files to change during execution.

3. Optional, but is there any reason this library isn't using asynchronous file reads? It seems maddening to me that this library has an Asynchronous API, but chooses to use the synchronous file operations under the hood?

I'm happy to put in a PR to make these improvements.

Further details of what our memory ends up being filled with because of this:

![image](https://user-images.githubusercontent.com/11363036/123461921-9808bf00-d5e1-11eb-96a8-e0fb89920fbb.png)
![image](https://user-images.githubusercontent.com/11363036/123461926-9b9c4600-d5e1-11eb-9934-e5a6dd1e6ac7.png)

Memory usage before disabling datadog fs plugin
![image](https://user-images.githubusercontent.com/11363036/123461996-af47ac80-d5e1-11eb-91e7-c70e91d1c9b0.png)

Memory usage after disabling
![image](https://user-images.githubusercontent.com/11363036/123462034-b79fe780-d5e1-11eb-99c1-ba8691496d8b.png)

**Comment by freshollie** [comment]
![image](https://user-images.githubusercontent.com/11363036/123461348-e9fd1500-d5e0-11eb-8308-c1a8e388c5e1.png)
![image](https://user-images.githubusercontent.com/11363036/123461369-f08b8c80-d5e0-11eb-8797-5c4fb40d532c.png)

**Comment by gajus** [comment]
Thanks – These sound like valid changes. PR would be welcome.

**Comment by freshollie** [comment]
Thanks for merging. Any ideas on roadmap for getting this bumped in Slonik?

**Comment by gajus** [comment]
It has been released and added to Slonik.
